app/models/concerns/host_common.rb

```diff
@@ -204,6 +204,24 @@ def available_puppetclasses
     environment.puppetclasses - parent_classes
   end
 
+  def inherited_params(include_source = false)
+    hp = {}
+    params = host_inherited_params_objects
+    params.each do |param|
+      case param
+        when CommonParameter
+          hp.update(Hash[param.name => include_source ? {:value => param.value, :source => N_('common').to_sym, :safe_value => param.safe_value} : param.value])
+        when GroupParameter
+          hp.update(Hash[param.name => include_source ? {:value => param.value, :source => N_('hostgroup').to_sym, :safe_value => param.safe_value, :source_name => hostgroup.title} : param.value])
+        else
+          source = param.class.to_s.gsub('Parameter', '').downcase
+          source = 'operatingsystem' if source == 'os'
```

Comment:
Hm okay, please file it as a separate issue then, perhaps it was refactored and broken earlier!

It should probably be checked for all possible types, or we move the string extraction into the parameter subclasses themselves, into some method.